fw/procfs.c

```diff
@@ -85,17 +106,42 @@ static int
 tfw_perfstat_seq_show(struct seq_file *seq, void *off)
 {
 #define SPRNE(m, e)	seq_printf(seq, m": %llu\n", e)
+#define SPRNED(m, e)	seq_printf(seq, m": %dms\n", e)
 #define SPRN(m, c)	seq_printf(seq, m": %llu\n", stat.c)
 
+	int i;
 	TfwPerfStat stat;
 	u64 serv_conn_active, serv_conn_sched;
 	SsStat *ss_stat = kmalloc(sizeof(SsStat) * num_online_cpus(),
 				  GFP_KERNEL);
+	unsigned int val[T_PSZ] = { 0 };
+	TfwPrcntlStats pstats = {.val = val};
+
 	if (!ss_stat)
 		T_WARN("Cannot allocate sync sockets statistics\n");
 
 	memset(&stat, 0, sizeof(stat));
+
+	if (health_stat_codes) {
+		stat.hm = kmalloc(tfw_hm_stats_size(health_stat_codes->ccnt),
+				  GFP_KERNEL);
+		if (stat.hm)
+			tfw_hm_stats_clone(stat.hm, health_stat_codes);
+	} else {
+		stat.hm = NULL;
```

Comment:
Why do we need this explicit assigning of NULL? Above we `memset` to zero this structure. I believe we can get rid of explicit `memset` using `TfwPerfStat stat = {0}`

Reply:
Hm, probably it's the bug from the time when this code was above the memset. I don't remember actually, but fixed.